ambry-store/src/main/java/com/github/ambry/store/BlobStoreStats.java

```diff
@@ -1184,6 +1200,12 @@ public void run() {
           }
           // The remaining index entries in keyFinalStates are DELETE tombstones left by compaction (whose associated PUT is not found)
           updateDeleteTombstoneStats(keyFinalStates.values());
+          // Remove delete tombstones from scan results
+          for(IndexFinalState state: keyFinalStates.values()){
+            if(state.isDelete() && state.getExpirationTime() != Utils.Infinite_Time){
```

Comment:
same here.

Reply:
See previous comment. Hopefully it makes some sense.